src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -478,14 +470,27 @@ const WorkflowView = _.flow(
         !isRedacted ? filterConfigIO(inputsOutputs) : _.identity
       )(config)
 
+      const selectedSnapshotEntityMetadata = modifiedConfig.dataReferenceName ? await Ajax(signal).Workspaces.workspace(namespace, name).snapshotEntityMetadata(namespace, modifiedConfig.dataReferenceName) : {}
+      const selectedSnapshotTableNames = _.keys(selectedSnapshotEntityMetadata)
+
       this.setState({
         savedConfig: config, modifiedConfig,
         currentSnapRedacted: isRedacted, savedSnapRedacted: isRedacted,
         entityMetadata,
+        availableSnapshots: snapshots,
+        selectedEntitySource: modifiedConfig.dataReferenceName ? 'snapshot' : 'table',
+        selectedTableName: modifiedConfig.dataReferenceName ? modifiedConfig.rootEntityType : undefined,
+        cachedSnapshotEntityMetadata: modifiedConfig.dataReferenceName ? [{ snapshotName: modifiedConfig.dataReferenceName, metadata: selectedSnapshotEntityMetadata }] : [],
```

Comment:
Why not simplify the structure of this state to this:
```suggestion
        cachedSnapshotEntityMetadata: modifiedConfig.dataReferenceName ? { [modifiedConfig.dataReferenceName]: selectedSnapshotEntityMetadata } : {},
```

Then, you won't have to `_.find` to check if something is there, just see if there's a value for the key.